### ISSUE TITLE
Application must be instantiated before call to window constructor due to resources in App.xaml

### ISSUE THREAD

**Issue by cmeeren** [elmish/Elmish.WPF#41]
When using `Program.runWindow`, the window must necessarily be instantiated first. Since `Application` isn't instantiated before calling `runWindow`, I get the following exception on the `InitializeComponent()` call in the window constructor:

`Exception: Cannot find resource named 'MaterialDesignMultiFloatingActionAccentPopupBox'. Resource names are case sensitive.`

This error occurs because the resource in question is defined in my `App.xaml`:

```xaml
<Application.Resources>
  <ResourceDictionary>
    <ResourceDictionary.MergedDictionaries>
      <ResourceDictionary Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesignTheme.Light.xaml" />
      <ResourceDictionary Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesignTheme.Defaults.xaml" />
      <ResourceDictionary Source="pack://application:,,,/MaterialDesignColors;component/Themes/Recommended/Primary/MaterialDesignColor.Blue.xaml" />
      <ResourceDictionary Source="pack://application:,,,/MaterialDesignColors;component/Themes/Recommended/Accent/MaterialDesignColor.Lime.xaml" />
      <ResourceDictionary Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesignTheme.PopupBox.xaml" />
    </ResourceDictionary.MergedDictionaries>
  </ResourceDictionary>
</Application.Resources>
```

Since this is not instantiated/loaded before the window is initialized, I am unable to initialize my main window to start the app.

What is the best way to solve this? Is there something I can do, or must this be fixed in Elmish.WPF?

**Comment by cmeeren** [comment]
One workaround is of course to define the resources in all XAML files, but that's a horrible solution for obvious reasons. Surely there must be some way to use application-wide resources when using Elmish.WPF?

If this is indeed impossible, the fix could be as simple as adding a function `runWithWindow` that instead of a `Window` takes `unit -> Window`, and use this in `Elmish.WPF.Program._run`, and instantiate `Application` before instantiating the window.

**Comment by cmeeren** [comment]
In fact, I see that Elmish.WPF calls `Application()` directly and doesn't use my custom `Application` subclass, which contains overrides for `OnStartup` etc. Is there any way to tell Elmish.WPF to use a custom `Application` class? If not: Could this be implemented, and are there any other ways to do `OnStartup` and similar stuff?

**Comment by cmeeren** [comment]
Closed by #48
